src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,142 @@
 package core.basesyntax;
 
+import static java.lang.Math.abs;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private HashObject<K, V>[] hashMap;
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int size = 0;
+
+    public MyHashMap() {
+        hashMap = new HashObject[DEFAULT_INITIAL_CAPACITY];
+    }
+
+    public K[] keySet() {
+        if (size == 0) {
+            return null;
+        } else {
+            K[] keys = (K[]) new Object[size];
+            int index = 0;
+            for (HashObject<K, V> object : hashMap) {
+                if (object != null) {
+                    HashObject<K,V> currnetNode = object;
+                    do {
+                        keys[index] = (K) currnetNode.key;
+                        currnetNode = currnetNode.next;
+                        index++;
+                    } while (currnetNode != null);
+                }
+            }
+            return keys;
+        }
+    }
+
+    public boolean containsKey(K key) {
+        if (size == 0) {
+            return false;
+        }
+        int ifItWasFound = 0;
+        for (K currentKey : keySet()) {
+            if ((key != null && currentKey != null && currentKey.equals(key))
+                    || key == null && currentKey == null) {
+                ifItWasFound++;
+            }
+        }
+        return ifItWasFound != 0;
+    }
+
+    public V[] values() {
+        V[] values = (V[]) new Object[size];
+        return values;
+    }
+
+    private HashObject getHashObject(K key) {
```

Comment:
The `getHashObject` method should check if the `currentNode` is `null` before accessing its `key` field. Add a null check at the beginning of the while loop to prevent a `NullPointerException`.